applications/sim_showers_for_trigger_rates.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python3
+
+'''
+    Summary
+    -------
+    This application simulates showers to be used in trigger rate calculations.
+    Arrays with one (1MST) or four telescopes (4LST) can be used, in case of \
+    mono or stereo trigger configurations, respectively. 
+
+    Simulations are managed by the shower_simulator module.
+    Each run is simulated in a job. Each job is submitted by using the submission \
+    command from the global config settings (see config_template.yml). \
+    The config entry extraCommands can be used to extra commands to be ran in each job,
+    before the actual simulation.
+
+    At the moment, the shower simulations are performed by CORSIKA, which requires \
+    the zstd package. Please, make sure that the command to set your zstd path is \
+    properly set by the extraCommands in config.yml.
+
+    Command line arguments
+    ----------------------
+    array (str, required)
+        Name of the array (1MST, 4LST ...).
+    site (str, required)
```

Comment:
All the array data files (e.g., data/layout/telescope_positions-North-1LST.ecsv) are for a given site. Why do need to give additional the site as an input parameter?

Reply:
We need the site and the layout name to build a LayoutArray.  I don't understand the issue.

Reply:
That when the array is given, the site is known. Not a problem, but the same information is entered twice.

Reply:
No, you need site and layoutName to define a LayoutArray.
The layoutName is '4LST', 'Prod5', ...
You can have a '4LST' array for both South and North.

Maybe the confusion comes from naming the argument 'array' and not 'array_name' or 'layout_name'.

Reply:
I think I am a bit lost here what happens:

- choosing 4LSTs + North means that the layout 'telescope_positions-North-4LST' is read (right now from a the file telescope_positions-North-4LST.ecsv, probably in future from the data base). What if I want to have another layout for 4 LSTs in the North? Are code changes necessary, or does the code builds e.g. telescope_positions-North-4LST-New.ecsv when I give array="4LST-New"?
- I think I am fine for the site parameter, as we are deriving other parameters (e.g., magnetic field) from the site parameter.

Reply:
> * choosing 4LSTs + North means that the layout 'telescope_positions-North-4LST' is read (right now from a the file telescope_positions-North-4LST.ecsv, probably in future from the data base). What if I want to have another layout for 4 LSTs in the North? Are code changes necessary, or does the code builds e.g. telescope_positions-North-4LST-New.ecsv when I give array="4LST-New"?

The code doesn't build the new layout at this level. You have to use another application (or write your own [1]) to build a new layout and then come back to this application, once your new layout is pre-defined.
The name 4LST is just a name, it doesn't mean anything for the code. We could have used instead 'array-with-4-LST-to-simulate-trigger-rates'. So it's our convention naming the regular arrays like this (following the sim_telarray convention). 

The idea is that building a new layout is a 'serious' action and should be done separately from other actions. We won't be building new layouts all the time. Most of the time we will be using pre-defined layouts that have been built in the past.

[1] As it's now (using ecsv files to store the layouts), one can just take an existing layout file, change its name and change the values by hand in order to create a new layout. That would work too.

Reply:
(resolved, as I understand now better what is happening)